fyle_accounting_mappings/views.py

```diff
@@ -317,42 +326,57 @@ def get_queryset(self):
         mapped = self.request.query_params.get('mapped')
         destination_type = self.request.query_params.get('destination_type', '')
 
+        # Process the 'mapped' parameter
         if mapped and mapped.lower() == 'false':
             mapped = False
         elif mapped and mapped.lower() == 'true':
             mapped = True
         else:
             mapped = None
 
+        # Prepare filters based on destination_type
         filters = {}
 
         if destination_type == 'ACCOUNT':
             filters['destination_account__attribute_type'] = destination_type
         else:
             filters['destination_expense_head__attribute_type'] = destination_type
 
+        # Get source categories
         source_categories = CategoryMapping.objects.filter(
             **filters,
             source_category__active=True,
             workspace_id=self.kwargs['workspace_id'],
         ).values_list('source_category_id', flat=True)
 
-        filters = {
-            'workspace_id' : self.kwargs['workspace_id'],
-            'attribute_type': 'CATEGORY',
-            'active': True
-        }
+        # Prepare filters for ExpenseAttribute
+        base_filters = Q(workspace_id=self.kwargs['workspace_id']) & \
+                       Q(attribute_type='CATEGORY') & \
+                       Q(active=True)
+
+        # Get the 'Activity' mapping and attribute
+        activity_mapping = CategoryMapping.objects.filter(
+            source_category__value='Activity', workspace_id=self.kwargs['workspace_id']).first()
+        activity_attribute = ExpenseAttribute.objects.filter(
+            attribute_type='CATEGORY', value='Activity', workspace_id=self.kwargs['workspace_id'], active=True).first()
+
+        # Adjust the filters if 'Activity' attribute exists but not mapped
+        if activity_attribute and not activity_mapping:
+            base_filters &= ~Q(value='Activity')
 
+        # Handle the mapped parameter
         param = None
-        if mapped:
+        if mapped is True:
```

Comment:
frontend query params will be parsed as string 'true', let the condition be as it is

Reply:
we are setting mapped above